Statistics.md

```diff
@@ -0,0 +1,85 @@
+
+### Bloom filter statistics
+
+Supported Types: Integer
+
+The bloom filter is stored per block which is used to filter some blocks when reading the parquet file.
+
+#### Background for bloom filter
+In short, bloom filter maintains a bit set which is set to 1 for all bits. Once one entry added to the filter, it will use the hash functions to set several bits to 1. Using thge same hash functions,
+we can see if a new entry is already included checking all of the related bits value. If all of them are set to 1, it means this entry is already added, otherwise not.
+In this way we can use the bloom filter to skip some data in block level for parquet. From the Wikipedia, you can see the definition of bloom filter as follows:
+```An empty Bloom filter is a bit array of m bits, all set to 0.
+There must also be k different hash functions defined, each of which maps or hashes some set element to one of the m array positions with a uniform random distribution.```
+To construct a bloom filter, two parameters are required:
+* _m_: Number of bits
+* _k_: Number of hash function
+These two parameters could be approximated by two other parameters (The formulas are available in the [Wikipedia](https://en.wikipedia.org/wiki/Bloom_filter)):
+
+* _f_: abbreviation for false positive probability
+* _n_: the number of elements being filtered
+
+The code snippet for the data structure of bloom filter is mainly referred from [Apache Hive](https://hive.apache.org/) project and the implementation of bloom filter from [Guava](http://code.google.com/p/guava-libraries/).
+
+### The structure of bloom filter in parquet-format
+In the statistics, we have the following information for bloom filter.
+```
+list<i64> bitSet: the bit set storing the generated bit values using added entries
+i32 numBits; number of bit which is evaluated in the way discussed in the previous section
+i32 numHashFunctions; number of hash functions to generate the bit from the encountering entries
+```
+
+#### Implementation of the bloom filter in parquet-mr
+Currently we support two bloom filter strategies. One is using two 32bit hash functions and another is using all 128 bits of Murmur3 128 when hashing. The first one is mentioned in "Less Hashing, Same Performance: Building a Better Bloom Filter" by Kirsch et.al. From abstract 'only two hash functions are necessary to effectively implement a Bloom filter without any loss in the asymptotic false positive probability' Lets split up 64-bit hashcode into two 32-bit hash codes and employ the technique mentioned in the above paper
+
+#### Build a bloom filter in the parquet-mr
+To build a bloom filter, users need to add the following four configurations:
+
+* _parquet.enable.bloom.filter_: Whether to enable the bloom filter
+* _parquet.bloom.filter.fpp.value_: The value of FPP (Abbr. false positive probability)
+* _parquet.bloom.filter.fpp.provided_: As the fpp is optional, so this configuration is not required and the default value is 0.05
```

Comment:
+1 let's simplify and just have an optional parquet.bloom.filter.fpp with a default of 0.05
(and put this in parquet-mr)